go/tools/builders/env.go

```diff
@@ -332,6 +336,40 @@ func absArgs(args []string, flags []string) {
 	}
 }
 
+
+// absToRelativePathWriter filters the output stream from a compile command,
+// replacing any absolute paths with relative equivalents.
+//
+// This is done on a best-effort basis, and not much effort is undertaken. If a
+// path is split across multiple Write calls, it will not be relativized.
+type absToRelativePathWriter struct {
+	pwd      string
+	delegate io.Writer
+}
+
+func (wr absToRelativePathWriter) Write(p []byte) (nn int, err error) {
+	if wr.pwd == "" {
+		return wr.delegate.Write(p)
+	}
+	if !strings.HasSuffix(wr.pwd, "/") {
+		wr.pwd += "/"
+	}
+	return wr.delegate.Write(bytes.ReplaceAll(p, []byte(wr.pwd), nil))
+}
+
+// rel converts absolute paths found in the given output string to relative, if
+// they are within the PWD.
+func rel(output string) string {
```

Comment:
Need a more descriptive name, maybe `relativizePaths`. I'd expect `rel` to work like `filepath.Rel` on a single path.

Reply:
Agreed, updated